### ISSUE TITLE
java.util.concurrent.ExecutionException: java.lang.UnsatisfiedLinkError: /tmp/libnetty-transport-native-epoll6734735497094555703.so: /tmp/libnetty-transport-native-epoll6734735497094555703.so: wrong ELF class: ELFCLASS64 (Possible cause: architecture word width mismatch)

### ISSUE THREAD

**Issue by brunoqc** [riemann/riemann#505]
Linux ubuntunew 3.13.0-44-generic #73-Ubuntu SMP Tue Dec 16 00:23:46 UTC 2014 i686 i686 i686 GNU/Linux

openjdk-7-jre-headless:i386           7u75-2.5.4-1~trusty1

```
bbigras@ubuntunew:~$ sudo -u riemann java -cp :/usr/share/riemann/riemann.jar riemann.bin start /etc/riemann/riemann.config
log4j:WARN No appenders could be found for logger (io.netty.util.internal.logging.InternalLoggerFactory).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
INFO [2015-01-29 13:42:33,078] main - riemann.bin - PID 27742
INFO [2015-01-29 13:42:33,419] clojure-agent-send-off-pool-3 - riemann.transport.websockets - Websockets server 0.0.0.0 5556 online
OpenJDK Client VM warning: You have loaded library /tmp/libnetty-transport-native-epoll6734735497094555703.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
INFO [2015-01-29 13:42:33,528] clojure-agent-send-off-pool-2 - riemann.transport.udp - UDP server 0.0.0.0 5555 16384 online
ERROR [2015-01-29 13:42:33,535] main - riemann.bin - Couldn't start
java.util.concurrent.ExecutionException: java.lang.UnsatisfiedLinkError: /tmp/libnetty-transport-native-epoll6734735497094555703.so: /tmp/libnetty-transport-native-epoll6734735497094555703.so: wrong ELF class: ELFCLASS64 (Possible cause: architecture word width mismatch)
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:188)
        at clojure.core$deref_future.invoke(core.clj:2180)
        at clojure.core$future_call$reify__6320.deref(core.clj:6420)
        at clojure.core$deref.invoke(core.clj:2200)
        at clojure.core$map$fn__4245.invoke(core.clj:2559)
        at clojure.lang.LazySeq.sval(LazySeq.java:40)
        at clojure.lang.LazySeq.seq(LazySeq.java:49)
        at clojure.lang.Cons.next(Cons.java:39)
        at clojure.lang.RT.next(RT.java:598)
        at clojure.core$next.invoke(core.clj:64)
        at clojure.core$dorun.invoke(core.clj:2856)
        at riemann.core$transition_BANG_.invoke(core.clj:156)
        at clojure.lang.Atom.swap(Atom.java:51)
        at clojure.core$swap_BANG_.invoke(core.clj:2233)
        at riemann.config$apply_BANG_.invoke(config.clj:293)
        at riemann.bin$_main.invoke(bin.clj:69)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at riemann.bin.main(Unknown Source)
Caused by: java.lang.UnsatisfiedLinkError: /tmp/libnetty-transport-native-epoll6734735497094555703.so: /tmp/libnetty-transport-native-epoll6734735497094555703.so: wrong ELF class: ELFCLASS64 (Possible cause: architecture word width mismatch)
        at java.lang.ClassLoader$NativeLibrary.load(Native Method)
        at java.lang.ClassLoader.loadLibrary1(ClassLoader.java:1965)
        at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1890)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1851)
        at java.lang.Runtime.load0(Runtime.java:795)
        at java.lang.System.load(System.java:1062)
        at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:193)
        at io.netty.channel.epoll.Native.<clinit>(Native.java:46)
        at io.netty.channel.epoll.EpollEventLoop.<init>(EpollEventLoop.java:70)
        at io.netty.channel.epoll.EpollEventLoopGroup.newChild(EpollEventLoopGroup.java:71)
        at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:64)
        at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:49)
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:56)
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:48)
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:41)
        at io.netty.channel.epoll.EpollEventLoopGroup.<init>(EpollEventLoopGroup.java:34)
        at riemann.transport.tcp$fn__4647.invoke(tcp.clj:79)
        at riemann.transport.tcp.TCPServer.start_BANG_(tcp.clj:132)
        at riemann.service$fn__1886$G__1880__1888.invoke(service.clj:20)
        at riemann.service$fn__1886$G__1879__1891.invoke(service.clj:20)
        at clojure.core$pmap$fn__6328$fn__6329.invoke(core.clj:6466)
        at clojure.core$binding_conveyor_fn$fn__4145.invoke(core.clj:1910)
        at clojure.lang.AFn.call(AFn.java:18)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by aphyr** [comment]
Haven't seen this one before; my guess is it's something to do with the native Netty epoll driver. Might be fixed in Netty upstream, or you might be able to work around it by turning off the use of epoll in riemann/transport/tcp.clj--there's already a conditional branch there for platform detection, so we could add an option to (tcp-server) to disable it as well.

**Comment by brunoqc** [comment]
I tried changing `4.0.24.Final` to `4.0.25.Final` in [project.clj](https://github.com/aphyr/riemann/blob/a5c5e3c0a30ff37fd35158a0e8ece78db470f0c3/project.clj#L33) and running `lein run` again withtout success (I'm not sure if it's enough to test with the latest Netty stable version).

It's fine if I disable epoll in riemann/transport/tcp.clj.

**Comment by pyr** [comment]
fixed in #508
